cypress/support/custom_commands/login.helper.js

```diff
@@ -0,0 +1,193 @@
+const nextButtonOnFirstLoginPages = '[id="nextSection"]'
+const skipToDashboardButtonOnFirstLoginPage =
+	'[data-testid="btn_schul-cloud_erkunden"]'
+const nbcLoginWithEmailOptionButton = '[data-testid="submit-cloud-site"]'
+const studentAgeSelectRadioBtn = '[id="reg-16"]'
+const nextButtonAfterAgeSelection = '#showExistingLoginForm'
+const datePickerSelectorForDOB =
+	'[data-testid="form-date-input-studentBirthdate"]'
+const studentUpdatePassword = '[data-testid="firstlogin_password"]'
+const studentConfirmPassword = '[data-testid="firstlogin_password_control"]'
+
+const env = Cypress.env()
+let environmentUpperCased
+
+const getUserCredentials = username => {
+	switch (username) {
+		case 'teacher1_brb':
+			return ['TEACHER_1_BRB_EMAIL', 'TEACHER_1_BRB_PASSWORD']
+		case 'teacher2_brb':
+			return ['TEACHER_2_BRB_EMAIL', 'TEACHER_2_BRB_PASSWORD']
+		case 'teacher1_dbc':
+			return ['TEACHER_1_DBC_EMAIL', 'TEACHER_1_DBC_PASSWORD']
+		case 'student1_brb':
+			return ['STUDENT_1_BRB_EMAIL', 'STUDENT_1_BRB_PASSWORD']
+		case 'student1_dbc':
+			return ['STUDENT_1_DBC_EMAIL', 'STUDENT_1_DBC_PASSWORD']
+		case 'admin1_brb':
+			return ['ADMIN_1_BRB_EMAIL', 'ADMIN_1_BRB_PASSWORD']
+		case 'teacher1_nbc':
+			return ['TEACHER_1_NBC_EMAIL', 'TEACHER_1_NBC_PASSWORD']
+		case 'student2_nbc':
+			return ['STUDENT_2_NBC_EMAIL', 'STUDENT_2_NBC_PASSWORD']
+		case 'admin1_nbc':
+			return ['ADMIN_1_NBC_EMAIL', 'ADMIN_1_NBC_PASSWORD']
+		case 'admin1_dbc':
+			return ['ADMIN_1_DBC_EMAIL', 'ADMIN_1_DBC_PASSWORD']
+		case 'student_extern_dbc':
+			return ['STUDENT_DBC_EXTERN', 'STUDENT_DBC_EXTERN_PASSWORD']
+		default:
+			return [null, null]
+	}
+}
+
+const performExternalLogin = () => {
+	cy.request('GET', oauth_url).then(resp => {
+		cy.intercept(resp.requestHeaders.referer).as('oauth_url')
+		cy.visit(resp.requestHeaders.referer).then(window => {
+			expect(window.location.pathname).to.include('/Account/Login')
+			cy.get('@oauth_url').then(resp => {
+				expect(resp.response.statusCode).to.equal(308)
+				cy.get(externalUsernameInputFieldElement).should('be.visible')
+				cy.get(externalUsernameInputFieldElement).type(env[userEmail], {
+					log: false
+				})
+				cy.get(externalPasswordInputFieldElement)
+					.type(env[userPassword], { log: false })
+					.type('{enter}')
+			})
+		})
+	})
+}
+
+const fillLoginForm = (username, password) => {
+	return (
+		cy.get(emailInputFieldElement).type(username, { log: true }),
+		cy.get(passwordInputFieldElement).type(password, { log: true }),
+		cy.get(submitButton).click()
+	)
+}
+
+const shuffleString = str => {
+	const characters = str.split('')
+	let currentIndex = characters.length
+	let randomIndex
+
+	while (currentIndex !== 0) {
+		randomIndex = Math.floor(Math.random() * currentIndex)
+		currentIndex--
+		;[characters[currentIndex], characters[randomIndex]] = [
+			characters[randomIndex],
+			characters[currentIndex]
+		]
+	}
+
+	return characters.join('')
+}
+const generateStrongPassword = length => {
```

Comment:
why this method is required? as we get initial password in the response to use.

Reply:
After login student needs to update the first password